### PR TITLE
feat(tests): add support for EROFS tests

### PR DESCRIPTION
**Pull request by musikid**
The execution of these tests is conditioned to the `allow_erofs` flag, since they require to remount the file system as read-only.

### CODE REVIEW

File: rust/src/tests/errors/erofs.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    #[cfg(any(
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "openbsd",
+        target_os = "dragonfly"
+    ))]
+    cmd.arg("-u");
+
+    // XXX: It is possible to do it directly with mount(2) but we use the CLI for now
```
Comment (asomers) [inline rust/src/tests/errors/erofs.rs:47]: Pro tip: it's basically impossible to use mount(2) on FreeBSD.  You should use nmount(2) instead.

Comment (musikid) [reply rust/src/tests/errors/erofs.rs:47]: Hmm, it doesn't look much harder to implement than mount(2). Should I implement it before merging?

Comment (asomers) [reply rust/src/tests/errors/erofs.rs:47]: Sure, if you think you can do it easily.  Nix already has bindings for nmount.

File: book/src/configuration-file.md [hunk 2]
```diff
@@ -77,8 +77,12 @@ entries = [
 ```toml
 [settings]
 naptime = 0.001
+allow_erofs = false
```
Comment (asomers) [inline book/src/configuration-file.md:80]: It looks like you forgot to update the documentation with the option's new name.
```suggestion
allow_remount = false
```

File: book/src/configuration-file.md [hunk 3]
```diff
@@ -77,8 +77,12 @@ entries = [
...
 ```
 
 - `naptime` - The duration for a "short" sleep. It should be greater than the
   timestamp granularity of the file system under test. The default value is 1
   second.
+- `allow_erofs` - If set to `true`, the runner will run the EROFS tests,
```
Comment (asomers) [inline book/src/configuration-file.md:86]:
```suggestion
- `allow_remount` - If set to `true`, the runner will run the EROFS tests,
```

File: rust/src/tests/errors/erofs.rs [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    let mount_result = cmd.arg("-o").arg(opt).arg(mountpoint.as_ref()).output()?;
```
Comment (asomers) [inline rust/src/tests/errors/erofs.rs:48]: Where is `opt` defined on non-Linux?

File: rust/src/utils.rs [hunk 5]
```diff
@@ -47,6 +49,29 @@ pub fn symlink<P: ?Sized + nix::NixPath>(path1: &P, path2: &P) -> nix::Result<()
     symlinkat(path1, None, path2)
 }
 
+/// Get mountpoint.
+pub fn get_mountpoint(base_path: &Path) -> Result<&Path, anyhow::Error> {
+    let base_dev = lstat(base_path)?.st_dev;
```
Comment (asomers) [inline rust/src/utils.rs:54]: On FreeBSD, you can do this much easier by using `statfs` and looking at the `f_mntonname` field.  I'm not sure what the best way to do it on Linux is.

**Review by asomers — CHANGES_REQUESTED** [review]

File: rust/src/utils.rs (hunk 5 above)
Comment (musikid) [reply rust/src/utils.rs:54]: Just above, although it isn't very clear.

File: rust/src/tests/errors/erofs.rs (hunk 4 above)
Comment (musikid) [reply rust/src/tests/errors/erofs.rs:48]: Just above, although it isn't very clear.

File: rust/src/utils.rs (hunk 5 above)
Comment (musikid) [reply rust/src/utils.rs:54]: Yeah I choose this way of doing it to ensure that it would work on multiple platforms.

**Review by asomers — APPROVED** [review]

**Pull request merged**